pkg/storage/mvcc.go

```diff
@@ -4955,20 +4953,16 @@ func MVCCGarbageCollect(
 			// We need to iterate ranges only to compute GCBytesAge if we are updating
 			// stats.
 			//
-			// TODO(erikgrinaker): Rewrite to use MVCCRangeKeyStack.
-			if _, hasRange := iter.HasPointAndRange(); hasRange && !lastRangeTombstoneStart.Equal(iter.RangeBounds().Key) {
-				rangeKeys := iter.RangeKeys()
-				lastRangeTombstoneStart = append(lastRangeTombstoneStart[:0], rangeKeys.Bounds.Key...)
-				rangeTombstoneTss = rangeTombstoneTss[:0]
-				for _, v := range rangeKeys.Versions {
-					rangeTombstoneTss = append(rangeTombstoneTss, v.Timestamp)
+			// We can rely on range key changed iterator functionality here as we do
```

Comment:
```suggestion
			// We can't rely on range key changed iterator functionality here as we do
```

Reply:
`RangeKeyChanged()` does fire correctly on seek operations too, but one has to take care to track it for every positioning operation. I'm fine with not doing it now though, this is simple and more obviously correct.

Reply:
I think tracking for all Next() is a non-starter here for now. We rely on get metadata to seek and potentially Next() as a side effect +all the subsequent iterations. We'll have to `hasChanged |= iter.RangeKeyChaneged()` through the whole method till it reaches this point. That would be too fragile even if we know that key doesn't change while iterating key's versions.